ROADMAP.md

```diff
@@ -8,60 +8,57 @@ enhancement process, please visit the enhancements [repository](https://github.c
 
 This document tracks only major changes done to Keylime. 
 
-## Rust Agent
-#### End of Q3-2022
-**Status:** mostly complete
-
-The Keylime agent is being ported to Rust Lang. This decision was made based on
-several reasons:
-
-* Rust is statically linked and does not require the ability to retrieve
-  dependencies. (Important for non internet connected machines or immutable read
-  only operating systems)
-* Rust can be more performant and generally requires less resources.
-* Rust provides strong safety guarantees (memory safe).
-
-For further details of development, please consult the [rust-keylime](https://github.com/keylime/rust-keylime)
-repository and the [Rust agent roadmap issue](https://github.com/keylime/keylime/issues/986).
-
 ## Removal of the Python Agent
-#### End of Q4-2022
+#### Q1-2023
 
 Once the Rust agent is stable the Python agent will be removed in two stages:
 
 1. Add deprecation warning and keep it for one major release
 2. Remove the agent from the code with the release of 7.0.0
 
 ## User Experience Improvements
-#### End of Q4-2022
 
-Some aspects of Keylime of the Keylime user experience can be improved: 
+Some aspects of Keylimes user experience can be improved: 
 
 * Update the user documentation (https://github.com/keylime/keylime/issues/1035)
-* Simplify the TLS setup (https://github.com/keylime/enhancements/pull/73)
-* Simplify the configuration (https://github.com/keylime/enhancements/pull/73)
-* Remove or rework WebUI
 * Investigate integration with monitoring systems (e.g. Prometheus)
 
-## Push Model
-#### Q4-2022 or Q1-2023
 
-Instead of the verifier connecting to the agent to retrieve the attestation data, 
-the agent can also send this data periodically to the verifier.
+## Move Keylime to a Plugin Architecture and Refactoring
+### Q2/Q3-2023
+
+The main eventloop in Keylime is very focused on TPM based attestation in combination with IMA and Measured Boot.
+This has the disadvantage that it is currently not easy to support other forms of claims and evidence 
+(e.g. Intel SGX, AMD SEV) and their retrieval via different methods (e.g. push model, one shot attestation).
+Moving to a more flexible plugin architecture allows us to implement those changes without requiring core changes to Keylime. 
+
+This entails the following aspects:
+* Evaluating the use of general policy languages for validation (e.g. Rego or Seedwing)
+* Use common remote attestation terminology (see where the current one differs to the [rats](https://datatracker.ietf.org/wg/rats/about/) one)
```

Comment:
As for the evaluation of use of general policy languages I would suggest to first collect evidence of shortcomings of our current policy language (we just changed it) before going to anything new. We first need to have a clear understanding why we are changing before we are changing it.

Reply:
@stefanberger yes agree. I think there is a difference between the IMA and Measured Boot validation here and where we put the decision making part in Keylime.
The IMA policy format is static, now well defined and covers most of what IMA currently offers. On the Measured Boot side we have a custom DSL and the need for very dynamic configurations which is currently not really user friendly.

If we want to unify all the policy evaluation into one layer, then we probably have to unify the format into one.

Probably @galmasi comment more on the current drawbacks and experiments with changing that.

Reply:
Yeah, maybe measured boot policy should also go into our runtime policy.

Reply:
... although you could have a bunch of machines that all run the same or similar BIOS but then need different IMA policies. In this case you may want to keep it separate but then need the runtime policy commands to work separately work measured boot.

Reply:
for mb policy, maybe we can use something like jsonpath (https://pypi.org/project/jsonpath-ng/)

Reply:
The fact we have an entirely custom DSL for Measured Boot policies (which I've helped to develop :-) ) which is really not user friendly is reason enough to at least look for alternatives. This is true especially in the case some of alternatives provide a general purpose mechanism that could potentially offload the actual policy processing - for IMA (runtime), MB (boot) and TPM - to an external component. Since we are promising/declaring we are about to embark in an **evaluation**, I cannot see what is objectionable here. Strong support it.